### PR TITLE
Port to dune

### PR DESCRIPTION
**Pull request by TheLortex**
Main post: mirage/mirage#969
Depends on https://github.com/mirage/mirage-os-shim/pull/7

With the new mirage tool we don't need to take mirage backends into account, C compilations flags are  managed automatically.

### CODE REVIEW

File: lib/cflags.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+flags=""
+if [ "$(uname -m)" = "x86_64" -o "$(uname -m)" = "x86" ]
```
Comment (emillon) [inline lib/cflags.sh]: Do you think it's possible to use `dune.configurator` for this?

File: mirage-entropy.opam [hunk 2]
```diff
@@ -8,36 +8,22 @@ author:       ["Hannes Mehnert" "David Kaloper" "Anil Madhavapeddy" "Dave Scott"
 maintainer:   "david@numm.org"
 license:      "BSD2"
 
-build: [ "ocaml" "pkg/pkg.ml" "build" "--pinned" "%{pinned}%" "--tests" "false"
-          "--with-mirage-xen" "%{mirage-xen:installed}%"
-          "--with-mirage-solo5" "%{mirage-solo5:installed}%"
-          "--with-ocaml-freestanding" "%{ocaml-freestanding:installed}%"
-]
+build: [ "dune" "build" "-p" "mirage-entropy"]
```
Comment (emillon) [inline mirage-entropy.opam]: I recommend using the "standard" dune-opam snippet with `subst` and `runtest` (even if there are no tests for now)

Comment (avsm) [reply mirage-entropy.opam]: fixed in f5201a0b48f8d4f2949d7c50364885282ae94bf4

**Comment by hannesm** [comment]
thanks for the inspiration, #47 was just merged, and now dune is used as build system for mirage-entropy

**Pull request closed without merging**